lib/android/app/src/main/java/com/wix/reactnativenotifications/core/notification/PushNotification.java

```diff
@@ -200,22 +200,43 @@ protected int createNotificationId(Notification notification) {
     }
 
     private void notifyReceivedToJS() {
-        mJsIOHelper.sendEventToJS(NOTIFICATION_RECEIVED_EVENT_NAME, mNotificationProps.asBundle(), mAppLifecycleFacade.getRunningReactContext());
+        if (mNotificationProps == null) {
+            Log.e(LOGTAG, "notifyReceivedToJS: mNotificationProps is null");
+            return;
+        }
+
+        Bundle bundle = mNotificationProps.asBundle();
+        if (bundle == null) {
+            Log.e(LOGTAG, "notifyReceivedToJS: asBundle is null");
+            return;
+        }
+
+        mJsIOHelper.sendEventToJS(NOTIFICATION_RECEIVED_EVENT_NAME, bundle, mAppLifecycleFacade.getRunningReactContext());
     }
 
     private void notifyReceivedBackgroundToJS() {
-        mJsIOHelper.sendEventToJS(NOTIFICATION_RECEIVED_BACKGROUND_EVENT_NAME, mNotificationProps.asBundle(), mAppLifecycleFacade.getRunningReactContext());
+        if (mNotificationProps == null) {
+            Log.e(LOGTAG, "notifyReceivedBackgroundToJS: mNotificationProps is null");
+            return;
+        }
+
+        Bundle bundle = mNotificationProps.asBundle();
+        if (bundle == null) {
+            Log.e(LOGTAG, "notifyReceivedBackgroundToJS: asBundle is null");
+            return;
+        }
+
+        mJsIOHelper.sendEventToJS(NOTIFICATION_RECEIVED_BACKGROUND_EVENT_NAME, bundle, mAppLifecycleFacade.getRunningReactContext());
     }
 
     private void notifyOpenedToJS() {
         Bundle response = new Bundle();
         try {
             response.putBundle("notification", mNotificationProps.asBundle());
+            mJsIOHelper.sendEventToJS(NOTIFICATION_OPENED_EVENT_NAME, response, mAppLifecycleFacade.getRunningReactContext());
         } catch (NullPointerException e) {
             Log.e(LOGTAG, "notifyOpenedToJS: Null pointer exception");
         }
```

Comment:
To be consistent could we change `notifyOpenedToJS` so that it is structured the same way as `notifyReceivedToJS` and `notifyReceivedBackgroundToJS` ☝️? Use early `return`s instead of `try/catch` 😄 

```suggestion
        if (mNotificationProps == null) {
            Log.e(LOGTAG, "notifyOpenedToJS: mNotificationProps is null");
            return;
        }

        Bundle bundle = new Bundle();
        if (bundle == null) {
            Log.e(LOGTAG, "notifyOpenedToJS: asBundle is null");
            return;
        }

        mJsIOHelper.sendEventToJS(NOTIFICATION_OPENED_EVENT_NAME, bundle, mAppLifecycleFacade.getRunningReactContext());
```

Reply:
Great suggestion!